tests/sycl/group_functions.cpp

```diff
@@ -0,0 +1,1800 @@
+HIPSYCL_KERNEL_TARGET
+T get_offset(size_t margin, size_t divisor = 1)
+{
+    using eT = elementType<T>;
+    if (std::numeric_limits<eT>::max() <= margin) {
+        return T{};
+    }
+    if constexpr (std::is_floating_point<eT>::value) {
+        return T{};
+    }
+
+    if constexpr (std::is_signed<T>::value) {
+        return TInit<T>(static_cast<eT>(std::numeric_limits<eT>::min() / divisor + margin));
+    }
+    return TInit<T>(static_cast<eT>(std::numeric_limits<eT>::max() / divisor - margin));
+}
+
+template<typename T>
+HIPSYCL_KERNEL_TARGET
+elementType<T> local_value(size_t id, size_t offsetBase)
+{
+  const size_t N = T{}.get_count();
+
+  auto offset = get_offset<elementType<T>>(offsetBase);
+  return static_cast<elementType<T>>(id + offset);
+}
+
+void create_bool_test_data(std::vector<char> &buffer, size_t local_size, size_t global_size)
+{
+  BOOST_REQUIRE(global_size == 4*local_size);
+  BOOST_REQUIRE(local_size+10 < 2*local_size);
+
+  // create host_buf 4 different possible configurations:
+  // 1: everything except one false
+  // 2: everything false
+  // 3: everything except one true
+  // 4: everything true
+
+  for(size_t i = 0; i < 2*local_size; ++i)
+      buffer[i] = false;
+  for(size_t i = 2*local_size; i < 4*local_size; ++i)
+      buffer[i] = true;
+
+  buffer[10] = true;
+  buffer[2*local_size+10] = false;
+
+  BOOST_REQUIRE(buffer[0] == false);
+  BOOST_REQUIRE(buffer[10] == true);
+  BOOST_REQUIRE(buffer[local_size] == false);
+  BOOST_REQUIRE(buffer[10+local_size] == false);
+  BOOST_REQUIRE(buffer[local_size*2] == true);
+  BOOST_REQUIRE(buffer[10+local_size*2] == false);
+  BOOST_REQUIRE(buffer[local_size*3] == true);
+  BOOST_REQUIRE(buffer[10+local_size*3] == true);
+}
+
+template<typename T, int LINE>
+void check_binary_reduce(std::vector<T> buffer, size_t local_size, size_t global_size, std::vector<bool> expected, std::string name, size_t break_size = 0) {
+  std::vector<std::string> cases{"everything except one false", "everything false", "everything except one true", "everything true"};
+
+  BOOST_REQUIRE(global_size/local_size == expected.size());
+  for(size_t i = 0; i < global_size/local_size; ++i) {
+    for(size_t j = 0; j < local_size; ++j){
+      // used to stop after first subgroup
+      if (break_size != 0  && j == break_size)
+          break;
+
+      T computed = buffer[i*local_size+j];
+      T expectedValue = TInit<T>(expected[i]);
+
+      BOOST_TEST(TCompare(expectedValue, computed), LINE << ":" << T2str(computed) << " at position " << j << " instead of " << T2str(expectedValue) << " for case: " << cases[i] << " " << name);
+
+      if (!TCompare(expectedValue, computed))
+        break;
+    }
+  }
+}
+
+}
+
+
+BOOST_FIXTURE_TEST_SUITE(group_functions_tests, reset_device_fixture)
+
+template<int N, int M, typename T>
+class test_kernel;
+
+template<int TEST_ID, typename T, typename DataGenerator, typename FunctionUnderTest, typename TestFunction>
```

Comment:
Why is `TEST_ID` capitalized?

Reply:
missed it when renaming the other parameters, also changed it to `CallingLine`, since that's what I'm using as 'id' anyway.